hadoop-hdds/rocksdb-checkpoint-differ/src/main/java/org/apache/ozone/rocksdb/util/ManagedSstFileReader.java

```diff
@@ -89,16 +90,31 @@ public long getEstimatedTotalKeys() throws RocksDBException {
   }
 
   public Stream<String> getKeyStream() throws RocksDBException {
+    return getKeyStream(Optional.empty(), Optional.empty());
+  }
+
+  public Stream<String> getKeyStream(Optional<String> lowerBound,
+      Optional<String> upperBound) throws RocksDBException {
     // TODO: [SNAPSHOT] Check if default Options and ReadOptions is enough.
     final MultipleSstFileIterator<String> itr =
         new MultipleSstFileIterator<String>(sstFiles) {
           private ManagedOptions options;
           private ReadOptions readOptions;
 
+          private Optional<ManagedSlice> lowerBoundSLice;
```

Comment:
nit: It is not general practice to use `Optional` as class parameter (even IDE shows it as warning). If it is done this way just to avoid null check when closing them, I would suggest to use `IOUtils#closeQuietly()`. 

There are many article about to not use `Optional` as class parameter or method parameter. Few of them are as:
* https://stackoverflow.com/questions/31922866/why-should-java-8s-optional-not-be-used-in-arguments
* http://dolszewski.com/java/java-8-optional-use-cases/
* https://rules.sonarsource.com/java/RSPEC-3553/

Reply:
done removed the use of optional from all the places.